pysumma/pysumma/Decisions.py

```diff
@@ -0,0 +1,124 @@
+class Decisions:
+    def __init__(self, filepath):
+        # self.path = path
+        # self.filename = filename
+        self.filepath = filepath
+        self.simulStart = simul_datetime('simulStart', self.filepath)
+        self.simulFinsh = simul_datetime('simulFinsh', self.filepath)
+        self.soilCatTbl = option('soilCatTbl', self.filepath)
+        self.vegeParTbl = option('vegeParTbl', self.filepath)
+        self.soilStress = option('soilStress', self.filepath)
+        self.stomResist = option('stomResist', self.filepath)
+        self.num_method = option('num_method', self.filepath)
+        self.fDerivMeth = option('fDerivMeth', self.filepath)
+        self.LAI_method = option('LAI_method', self.filepath)
+        self.f_Richards = option('f_Richards', self.filepath)
+        self.groundwatr = option('groundwatr', self.filepath)
+        self.hc_profile = option('hc_profile', self.filepath)
+        self.bcUpprTdyn = option('bcUpprTdyn', self.filepath)
+        self.bcLowrTdyn = option('bcLowrTdyn', self.filepath)
+        self.bcUpprSoiH = option('bcUpprSoiH', self.filepath)
+        self.bcLowrSoiH = option('bcLowrSoiH', self.filepath)
+        self.veg_traits = option('veg_traits', self.filepath)
+        self.canopyEmis = option('canopyEmis', self.filepath)
+        self.snowIncept = option('snowIncept', self.filepath)
+        self.windPrfile = option('windPrfile', self.filepath)
+        self.astability = option('astability', self.filepath)
+        self.canopySrad = option('canopySrad', self.filepath)
+        self.alb_method = option('alb_method', self.filepath)
+        self.compaction = option('compaction', self.filepath)
+        self.snowLayers = option('snowLayers', self.filepath)
+        self.thCondSnow = option('thCondSnow', self.filepath)
+        self.thCondSoil = option('thCondSoil', self.filepath)
+        self.spatial_gw = option('spatial_gw', self.filepath)
+        self.subRouting = option('subRouting', self.filepath)
+
+class option:
+    def __init__(self, name, filepath):
+        self.filepath = filepath
+        self.open_read()
+        self.name = name
+        self.get_description()
+        self.options = self.get_options()
+
+    def open_read(self):
```

Comment:
Doesn't this end up reading the decisions file every time a new `option` class instantiation occurs? It seems that since the `Decision` class is a one-to-one mapping with the decisions file that reading/writing operations should go in that class.

Reply:
@arbennett Each time an `Option` object's `value` attribute is changed, the new value has to be written to the corresponding txt file to keep it up to date. To do this, a setter method is needed. Rather than write basically identical setter methods for each option, we made it a class and wrote those methods in the class. 
So you are right. Every time an `Option" object is instantiated the file is read, but we couldn't think of an easier way to do it with the interactions with the text file which can change based on user actions. Additionally, the file io is pretty quick, so I don't think performance will suffer at all. It works and I don't see it as something that would hold functionality back, but it seems like there would be a way to do it at the higher level. We are definitely open to ideas of how to move that functionality to the `Decision`/`Simulation` class for simplicity.